exporter/loadbalancingexporter/resolver_k8s.go

```diff
@@ -0,0 +1,301 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package loadbalancingexporter // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/loadbalancingexporter"
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"sort"
+	"strings"
+	"sync"
+
+	"go.opencensus.io/stats"
+	"go.opencensus.io/tag"
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/clientcmd"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+)
+
+var _ resolver = (*k8sResolver)(nil)
+
+var (
+	errNoSvc                        = errors.New("no service specified to resolve the backends")
+	errNoSvcPorts                   = errors.New("no ports specified to resolve the backends")
+	k8sResolverMutator              = tag.Upsert(tag.MustNewKey("resolver"), "k8s service")
+	k8sResolverSuccessTrueMutators  = []tag.Mutator{resolverMutator, successTrueMutator}
+	k8sResolverSuccessFalseMutators = []tag.Mutator{resolverMutator, successFalseMutator}
+)
+
+type k8sResolver struct {
+	logger    *zap.Logger
+	service   string
+	namespace string
+	port      []int32
+
+	handler        *handler
+	once           *sync.Once
+	epsListWatcher cache.ListerWatcher
+	endpointsStore *sync.Map
+
+	endpoints         []string
+	onChangeCallbacks []func([]string)
+
+	stopCh             chan (struct{})
+	updateLock         sync.Mutex
+	shutdownWg         sync.WaitGroup
+	changeCallbackLock sync.RWMutex
+}
+
+func newK8sResolver(clt kubernetes.Interface,
+	logger *zap.Logger,
+	service string,
+	ports []int32) (*k8sResolver, error) {
+
+	if len(service) == 0 {
+		return nil, errNoSvc
+	}
+	if len(ports) == 0 {
+		return nil, errNoSvcPorts
+	}
+
+	nAddr := strings.SplitN(service, ".", 2)
+	name, namespace := nAddr[0], "default"
+	if len(nAddr) > 1 {
+		namespace = nAddr[1]
+	}
```

Comment:
I would log at info level here, saying that no namespace was provided and that the namespace for this collector will be used (can we infer that from the in-cluster client?)

Reply:
Good idea! LB pods and Backends are most likely in the same ns, it should be more reasonable to fall back to this ns instead of `default`. But it am not sure whether ns can be obtained, I will check in detail.

Reply:
ns inference has been added and the priority is now:
user specified -> current collector ns -> fallback `default` (if infer fails)